projects/timer_challenge/src/main.c

```diff
@@ -0,0 +1,42 @@
+#include "interrupt.h"   // interrupts are required for soft timers
+#include "log.h"         // for printing
+#include "soft_timer.h"  // for soft timers
+#include "wait.h"        // for wait function
+
+#define COUNTER_PERIOD_MS 500
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void prv_timer_callback(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;  // cast void* to our struct so we can use it
+  storage->counter_a++;
+  LOG_DEBUG("Counter A: %i\n", storage->counter_a);
+
+  if (storage->counter_a % 2 == 0) {
+    storage->counter_b++;
+    LOG_DEBUG("Counter B: %i\n", storage->counter_b);
+  }
+
+  // start the timer again, so it keeps periodically incrementing every half a second
+  soft_timer_start_millis(COUNTER_PERIOD, prv_timer_callback, storage, NULL);
```

Comment:
you missed updating these values, so the automated build is failing since it doesn't compile